src/transports/websocket.ts

```diff
@@ -359,14 +359,19 @@ export class WebSocketTransport<
     // Send messages only if the connection is open
     // Otherwise we could encounter the case where we just closed the connection because there's no desired ones,
     // but without this check we'd attempt to send out all the unsubscribe messages
-    if (!connectionClosed && this.config.builders) {
+    if (!connectionClosed) {
       logger.debug('Connection is open, sending subs/unsubs if there are any')
-      const { subscribeMessage, unsubscribeMessage } = this.config.builders
-      await this.sendMessages(
-        context,
-        subscribeMessage ? subscriptions.new.map(subscribeMessage) : subscriptions.new,
-        unsubscribeMessage ? subscriptions.stale.map(unsubscribeMessage) : subscriptions.stale,
-      )
+      const builders = this.config.builders
+      if (builders) {
+        const { subscribeMessage, unsubscribeMessage } = builders
+        await this.sendMessages(
+          context,
+          subscribeMessage ? subscriptions.new.map(subscribeMessage) : subscriptions.new,
+          unsubscribeMessage ? subscriptions.stale.map(unsubscribeMessage) : subscriptions.stale,
+        )
+      } else {
+        await this.sendMessages(context, subscriptions.new, subscriptions.stale)
+      }
```

Comment:
As far as i remember there was a bug that's why i changed this logic. The builders are optional (makes sense since we can communicate with url only) but a WS test with no `builders` property would hang until timeout.

Reply:
Makes sense 👍